sdk/log/batch_test.go

```diff
@@ -4,15 +4,169 @@
 package log // import "go.opentelemetry.io/otel/sdk/log"
 
 import (
+	"context"
+	"slices"
 	"strconv"
+	"sync"
+	"sync/atomic"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	"go.opentelemetry.io/otel"
 )
 
+func TestBatchingProcessorEnabled(t *testing.T) {
+	b := NewBatchingProcessor(nil)
+	t.Cleanup(func() {
+		assert.NoError(t, b.Shutdown(context.Background()))
+	})
+	assert.True(t, b.Enabled(context.Background(), Record{}))
+}
+
+func TestBatchingProcessorShutdown(t *testing.T) {
+	e := new(exporter)
+	b := NewBatchingProcessor(e)
+	_ = b.Shutdown(context.Background())
+	require.True(t, e.shutdownCalled, "exporter Shutdown not called")
+}
+
+func TestBatchingProcessorForceFlush(t *testing.T) {
+	e := new(exporter)
+	b := NewBatchingProcessor(e)
+	_ = b.ForceFlush(context.Background())
+	require.True(t, e.forceFlushCalled, "exporter ForceFlush not called")
+}
+
+func TestBatchingProcessorShutdownCanceled(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+
+	b := NewBatchingProcessor(slowExporter{})
+	err := b.Shutdown(ctx)
+	require.ErrorIs(t, err, context.Canceled)
+}
+
+func TestBatchingProcessorForceFlushCancled(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+
+	b := NewBatchingProcessor(slowExporter{})
+	err := b.ForceFlush(ctx)
+	require.ErrorIs(t, err, context.Canceled)
+}
+
+func TestBatchingProcessorOnEmit(t *testing.T) {
+	e := new(exporter)
+	b := NewBatchingProcessor(e)
+	t.Cleanup(func() {
+		assert.NoError(t, b.Shutdown(context.Background()))
+	})
+
+	var r Record
+	r.SetSeverityText("test")
+	_ = b.OnEmit(context.Background(), r)
+	require.False(t, e.exportCalled, "exporter Export called")
+}
+
+func TestBatchingProcessorOnEmitForceFlush(t *testing.T) {
+	e := new(exporter)
+	b := NewBatchingProcessor(e)
+	t.Cleanup(func() {
+		assert.NoError(t, b.Shutdown(context.Background()))
+	})
+
+	var r Record
+	r.SetSeverityText("test")
+	_ = b.OnEmit(context.Background(), r)
+	_ = b.ForceFlush(context.Background())
+	require.True(t, e.exportCalled, "exporter Export not called")
+	assert.Equal(t, []Record{r}, e.records)
+}
+
+func TestBatchingProcessorOnEmitTick(t *testing.T) {
+	e := new(syncExporter)
+	s := NewBatchingProcessor(e, WithExportInterval(time.Millisecond))
+	t.Cleanup(func() {
+		assert.NoError(t, s.Shutdown(context.Background()))
+	})
+
+	var r Record
+	r.SetSeverityText("test")
+	_ = s.OnEmit(context.Background(), r)
```

Comment:
Missing test for premature flush of queue when the records are not the same age as the interval.